pkg/agent/cniserver/server_windows_test.go

```diff
@@ -0,0 +1,1038 @@
+				Error: &cnipb.Error{
+					Code:    cnipb.ErrorCode_CONFIG_INTERFACE_FAILURE,
+					Message: "failed to configure container IP: unable to attach HnsEndpoint",
+				},
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			isDocker := isDockerContainer(tc.netns)
+			testUtil := newHnsTestUtil(generateUUID(t), tc.existingHnsEndpoints, isDocker, tc.isAttached, tc.hnsEndpointCreatErr, tc.endpointAttachErr)
+			testUtil.setFunctions()
+			waiter := newAsyncWaiter(tc.podName, tc.infraContainerID)
+			server := mockCNIServer(t, controller, waiter.notifier)
+			requestMsg, ovsPortName := prepareSetup(t, ipamType, tc.podName, tc.containerID, tc.infraContainerID, tc.netns, nil)
+			if tc.endpointExists {
+				server.podConfigurator.ifConfigurator.(*ifConfigurator).addEndpoint(getHnsEndpoint(generateUUID(t), ovsPortName))
+			}
+			if tc.oriIPAMResult != nil {
+				ipam.AddIPAMResult(tc.infraContainerID, tc.oriIPAMResult)
+			}
+			if tc.ipamAdd {
+				ipamMock.EXPECT().Add(gomock.Any(), gomock.Any(), gomock.Any()).Return(true, oriIPAMResult, tc.ipamError).Times(1)
+			}
+			if tc.ipamDel {
+				ipamMock.EXPECT().Del(gomock.Any(), gomock.Any(), gomock.Any()).Return(true, nil).Times(1)
+			}
+			ovsPortID := generateUUID(t)
+			if tc.connectOVS {
+				if isDocker {
+					mockOVSBridgeClient.EXPECT().CreateInternalPort(ovsPortName, int32(0), gomock.Any(), gomock.Any()).Return(ovsPortID, nil).Times(1)
+					mockOVSBridgeClient.EXPECT().GetOFPort(ovsPortName, false).Return(int32(100), nil).Times(1)
+				} else {
+					mockOVSBridgeClient.EXPECT().CreatePort(ovsPortName, ovsPortName, gomock.Any()).Return(ovsPortID, nil).Times(1)
+					mockOVSBridgeClient.EXPECT().SetInterfaceType(ovsPortName, "internal").Return(nil).Times(1)
+					mockOVSBridgeClient.EXPECT().GetOFPort(ovsPortName, true).Return(int32(100), nil).Times(1)
+				}
+				mockOFClient.EXPECT().InstallPodFlows(ovsPortName, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
+				routeMock.EXPECT().AddLocalAntreaFlexibleIPAMPodRule(gomock.Any()).Return(nil).Times(1)
+			}
+			resp, err := server.CmdAdd(ctx, requestMsg)
+			assert.Equal(t, tc.expectedErr, err)
+			containerID := requestMsg.CniArgs.ContainerId
+			_, exists := ifaceStore.GetContainerInterface(containerID)
+			assert.Equal(t, exists, tc.containerIfaceExist)
+			if tc.response != nil {
+				assert.Equal(t, tc.response, resp)
+			} else if tc.expectedErr == nil {
+				cniResult := &current.Result{
+					CNIVersion: supportedCNIVersion,
+					IPs:        oriIPAMResult.IPs,
+					Routes:     oriIPAMResult.Routes,
+					DNS: cnitypes.DNS{
+						Nameservers: dns,
+						Search:      dnsSearches,
+					},
+					Interfaces: []*current.Interface{
+						{Name: ovsPortName, Mac: containerMACStr, Sandbox: ""},
+						{Name: "eth0", Mac: containerMACStr, Sandbox: tc.netns},
+					},
+				}
+				successResponse := resultToResponse(cniResult)
+				assert.Equal(t, successResponse, resp)
+			}
```

Comment:
Better to move 535 to 553 right after line 530, and merge the check with line 531.

Reply:
updated.